src/util/parse_options.cpp

```diff
@@ -35,16 +35,23 @@ void parse_options_baset::help()
 
 void parse_options_baset::usage_error()
 {
-  std::cerr << "Usage error!\n\n";
+  error_message("Usage error!\n");
```

Comment:
Why is passing a message handler to `parse_options_baset` not an option, so that this would be something like `log.error() << "Usage error!\n" << messaget::eom;`

Reply:
Various programs inherit from both `messaget` and `parse_option_baset`.  If you would prefer things like `program_parse_options::program_parse_options : messaget(), parse_option_baset(*this) { ...` then I can do it that way.

Reply:
Even though I strongly disagree with the inheritance done elsewhere, as far as `parse_options_baset` is concerned that's what I was thinking, yes. So I'd add a `message_handlert &message_handler;` member to `parse_options_baset` and then, on demand, create a `messaget log(message_handler);`. That way we can get away with a forward declaration of `class message_handlert` in `parse_options.h`, don't actually create unnecessary objects that we should not need under normal circumstances, and still do the output in the way desired and controlled by the caller.

Reply:
Hmm, I thought the difference should be fairly small?